src/converse-muc-views.js

```diff
@@ -884,12 +895,21 @@ converse.plugins.add('converse-muc-views', {
                                 this.model.get('jid'), args[0], 'participant', args[1],
                                 undefined, this.onCommandError.bind(this));
                         break;
+                    case 'destroy':
+                        if (!this.verifyAffiliations(['owner'])) {
+                            break;
+                        }
+                        this.destroy(
+                                this.model.get('jid'), args[0],
+                                this.onDestroySuccess.bind(this), this.onCommandError.bind(this));
```

Comment:
Instead of making a new method `onDestroySuccess`, you can simply use `this.close` here, and you can use an arrow function to avoid having to use `.bind`.

In `destroy`, you can simply return `_converse.api.sendIQ(iq)` so that it returns a promise.

So it could be something like this:
```
 this.destroy(this.model.get('jid'), args[0])
       .then(() => this.close())
       .catch(e => this.onCommandError(e));
```